core/store/src/db.rs

```diff
@@ -934,12 +942,12 @@ impl RocksDB {
         })
     }
 
-    pub fn new_read_only<P: AsRef<std::path::Path>>(path: P) -> Result<Self, DBError> {
-        RocksDBOptions::default().read_only(path)
+    pub fn new_read_only<P: AsRef<std::path::Path>>(path: P, store_config: &StoreConfig) -> Result<Self, DBError> {
+        RocksDBOptions::default().read_only(path, &store_config)
     }
 
-    pub fn new<P: AsRef<std::path::Path>>(path: P) -> Result<Self, DBError> {
-        RocksDBOptions::default().read_write(path)
+    pub fn new<P: AsRef<std::path::Path>>(path: P, store_config: &StoreConfig) -> Result<Self, DBError> {
+        RocksDBOptions::default().read_write(path, &store_config)
```

Comment:
This feels wrong. `StoreConfig` has a `read_only` flag so it feels those two methods should be merged into one which checks the config. `get_version` uses `new_read_only` so there may be reason to keep that one but it probably should be private.